src/Enzo/enzo_EnzoIntegrableUpdate.hpp

```diff
@@ -40,79 +39,91 @@ class EnzoIntegrableUpdate : public PUP::able
+  ///     will be added to fields held in this map. This should also contain
   ///     the passive scalars in specific form (as mass fractions) that are to
   ///     be integrated.
-  /// @param dUcons_group contains the fields where the net changes to the
+  /// @param[in]  dUcons_map Map of arrays where the net changes to the
   ///     integrable quantities and passively advected quantites are stored.
-  ///     If CT is being used, this will not include changes to the magnetic
-  ///     fields.
-  /// @param out_integrable_group contains the fields where the updated
-  ///     integrable quantities will be stored (This can be passed the same
-  ///     object as initial_integrable_group). The updated passively advected
-  ///     scalars will NOT be stored here.
-  /// @param out_conserved_passive_scalar contains the fields where the updated
+  ///     If constrained transport is being used, this will not include arrays
+  ///     for the magnetic fields.
+  /// @param[out] out_integrable_map Map of the fields where the updated
+  ///     integrable quantities will be stored (This can be a reference to the
+  ///     same Map referenced by initial_integrable_map). The updated passively
+  ///     advected scalars will NOT be stored here.
+  /// @param[out] out_conserved_passive_scalar Map of arrays where the updated
   ///     passive scalar quantities are stored in conserved form (as densities).
-  /// @param eos Pointer to the fluid's equation of state object. When
-  ///     applicable used for placing density and pressure floors and
-  ///     synchronizing the internal energy with the total energy
-  /// @param stale_depth The stale depth at the time of the function call
-  ///     (the stale_depth needs to be incremented after this function is
-  ///     called)
-  void update_quantities(Block *block, Grouping &initial_integrable_group,
-			 Grouping &dUcons_group,
-			 Grouping &out_integrable_group,
-			 Grouping &out_conserved_passive_scalar,
-			 EnzoEquationOfState *eos, int stale_depth) const;
-
-  /// provides a const vector of all integrable and passively advected scalars
-  const std::vector<std::string> combined_integrable_groups() const throw()
-  { return combined_integrable_groups_; }
+  /// @param[in]  eos Pointer to the fluid's equation of state object. When
+  ///     applicable used for placing a density floor.
+  /// @param[in]  stale_depth The stale depth at the time of the function call
+  ///     (the stale_depth must be incremented after this function is called)
+  /// @param[in]  passive_lists A list of lists of keys for passively advected
+  ///     scalars. The first list holds the keys for quantities that are
+  ///     normally passively advected. Subsequent lists group together
+  ///     collections pf passively advected scalars whose specific value
+  ///     (before and after advection) must sum to 1. Keys must not be
+  ///     duplicated across more than one list.
```

Comment:
> I'm asking because if this is a "sometime in the future" feature that may or may not be implemented by you, then it may be cleaner to remove that implicit logic and just go with a single list for now (and then make the update when additional lists will actually be supported).

Good point. I'll look into this. I think most of the pieces should be in place. It may be easier just to add this feature than to remove the logic.